packages/eslint-plugin/tests/rules/strict-boolean-expressions.test.ts

```diff
@@ -164,13 +180,13 @@ ruleTester.run('strict-boolean-expressions', rule, {
         if (!theEnum) {
         }
       `,
-      options: [{ allowNullableEnum: true }],
+      options: [{ allowEnum: true }],
     },
     {
       code: `
         enum ExampleEnum {
           This = 1,
-          That = 2,
+          That = 'two',
```

Comment:
This diff looks weird. I've moved the tests for `is('nullish', 'truthy number', 'enum')` and `is('nullish', 'truthy string', 'enum')` conditions to the `allowEnum` tests group and I've changed the option to be `allowEnum` instead of `allowNullableEnum`